File: app/src/main/java/com/shevaalex/android/rickmortydatabase/source/database/Converters.java
package com.shevaalex.android.rickmortydatabase.source.database;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.shevaalex.android.rickmortydatabase.models.character.LinkedLocationModel;

import java.lang.reflect.Type;

public class Converters {
    @TypeConverter
    public static String stringArrayToString(String[] array) {
        return new Gson().toJson(array);
    }

    @TypeConverter
    public static String[] stringToArray(String json) {
        Type arrayType = new TypeToken<String[]>(){}.getType();
        return new Gson().fromJson(json, arrayType);
    }

    @TypeConverter
    public static String linkedLocationModelToString(LinkedLocationModel locationModel) {
        return new Gson().toJson(locationModel);
    }

    @TypeConverter
    public static LinkedLocationModel stringToLinkedLocationModel(String string) {
        return new Gson().fromJson(string, LinkedLocationModel.class);
    }
}

File: app/src/main/java/com/shevaalex/android/rickmortydatabase/source/database/Converters.kt
package com.shevaalex.android.rickmortydatabase.source.database

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.shevaalex.android.rickmortydatabase.models.character.LinkedLocationModel

object Converters {
    @JvmStatic
    @TypeConverter
    fun listToString(list: List<String>): String {
        return Gson().toJson(list)
    }

    @JvmStatic
    @TypeConverter
    fun stringToList(json: String): List<String> {
        val listType = object : TypeToken<List<String>>() {}.type
        return Gson().fromJson(json, listType)
    }

    @JvmStatic
    @TypeConverter
    fun linkedLocationModelToString(locationModel: LinkedLocationModel): String {
        return Gson().toJson(locationModel)
    }

    @JvmStatic
    @TypeConverter
    fun stringToLinkedLocationModel(string: String): LinkedLocationModel {
        return Gson().fromJson(string, LinkedLocationModel::class.java)
    }
}